### PR TITLE
[bugfix] fix elasticsearch db

### PR DESCRIPTION
**Pull request by deven298**
## Description

This PR includes following changes:
- It skips embedding data when the content has not changed for the same source.
- Makes the query work with metadata filtering. For ex - 

```python
from embedchain import App

config = {
    "vectordb": {
        "provider": "elasticsearch",
        "config": {
            "collection_name": "ec-test",
            "cloud_id": "xxx",
            "basic_auth": ("elastic", "password"),
            "verify_certs": True,
        },
    }
}

app = App.from_config(config=config)

app.add("https://www.forbes.com/profile/elon-musk", metadata={"source": "forbes"})
app.add("https://en.wikipedia.org/wiki/Elon_Musk", metadata={"source": "wikipedia"})

response, context = app.query("which companies does elon own?", where={"source": "forbes"}, citations=True)
print(response)
print(list(map(lambda x: x[1]["url"], context)))
# The response will be coming from filtered documents where `source==forbes`.
```

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [x] Unit Test
- [x] Test Script

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] Made sure Checks passed

### CODE REVIEW

File: embedchain/vectordb/elasticsearch.py [hunk 1]
```diff
@@ -99,14 +99,24 @@ def get(self, ids: Optional[list[str]] = None, where: Optional[dict[str, any]] =
             query = {"bool": {"must": [{"ids": {"values": ids}}]}}
         else:
             query = {"bool": {"must": []}}
-        if "app_id" in where:
-            app_id = where["app_id"]
-            query["bool"]["must"].append({"term": {"metadata.app_id": app_id}})
 
-        response = self.client.search(index=self._get_index(), query=query, _source=False, size=limit)
+        if where is not None and len(where) > 0:
```
Comment (deshraj) [inline embedchain/vectordb/elasticsearch.py:103]:
```suggestion
        if where is not None:

```

File: embedchain/vectordb/elasticsearch.py [hunk 2]
```diff
@@ -186,9 +196,11 @@ def query(
                 },
             }
         }
-        if "app_id" in where:
-            app_id = where["app_id"]
-            query["script_score"]["query"] = {"match": {"metadata.app_id": app_id}}
+
+        if len(where) > 0:
```
Comment (deshraj) [inline embedchain/vectordb/elasticsearch.py:200]: Why checking based on the length? Why not do this:

```
if not where:
```

Comment (deven298) [reply embedchain/vectordb/elasticsearch.py:200]: You mean `if where:` will update it.

File: embedchain/vectordb/elasticsearch.py (hunk 1 above)
Comment (deven298) [reply embedchain/vectordb/elasticsearch.py:103]: Should be `if where:`

Comment (deven298) [inline embedchain/vectordb/elasticsearch.py:103]:
```suggestion
        if where:
```

File: embedchain/vectordb/elasticsearch.py (hunk 2 above)
Comment (deven298) [inline embedchain/vectordb/elasticsearch.py:200]:
```suggestion
        if where:
```

**Review by taranjeet — APPROVED** [review]
this looks good. thanks Deven.

**Pull request merged**